cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -215,6 +217,13 @@ func getControllerManagerCommand(cfg *kubeadmapi.MasterConfiguration, k8sVersion
 		"controllers":                      "*,bootstrapsigner,tokencleaner",
 	}
 
+	// If using external CA, pass empty string to controller manager instead of ca.key/ca.crt path,
+	// so that the csrsigning controller fails to start
+	if res, _ := certphase.UsingExternalCA(cfg); res {
+		defaultArguments["cluster-signing-key-file"] = ""
```

Comment:
Gah, should have spotted this earlier, I realize `delete(defaultArguments, "cluster...")` is probably preferable over this method... WDYT?

It would be great to get an unit test for this as well if possible...

Reply:
The nice thing about setting to "" is that (in controller manager logs) you get:

> Skipping "csrsigning"

instead of: 

> Failed to start certificate controller: error reading CA cert file "/etc/kubernetes/ca/ca.pem": open /etc/kubernetes/ca/ca.pem: no such file or directory

when using `delete()` which I think is slightly better, but I will switch them if you prefer.

Reply:
With that context (thanks for reminding me about the UX), it makes sense to pass `""`